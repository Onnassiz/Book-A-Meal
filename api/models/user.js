
module.exports = (sequelize, DataTypes) => {
	const User = sequelize.define('user', {
		id: {
			allowNull: false,
			primaryKey: true,
			type: DataTypes.UUID,
			defaultValue: DataTypes.UUIDV4,
		},
		email: {
			type: DataTypes.STRING,
			validate: {
				isEmail: true,
			},
		},
		passwordHash: DataTypes.STRING,
		fullName: DataTypes.STRING,
		role: DataTypes.STRING,
	}, {});
	User.associate = (models) => {
		User.hasMany(models.meal);
		User.hasMany(models.menu);
		User.hasOne(models.profile);
	};
	return User;
};
