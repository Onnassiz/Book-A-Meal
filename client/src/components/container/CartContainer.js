import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { deleteFromCart, updateCart } from '../../reducer/cart/action';
import Cart from '../presentation/Cart';

const mapStateToProps = (state) => {
  return {
    cart: state.cart,
    user: state.user,
    formState: state.formState,
  };
};

function mapDispatchToProps(dispatch) {
  return bindActionCreators({
    updateCart,
    deleteFromCart,
  }, dispatch);
}

export default withRouter(connect(
  mapStateToProps,
  mapDispatchToProps,
)(Cart));
