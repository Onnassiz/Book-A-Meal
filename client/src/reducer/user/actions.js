import axios from 'axios';
import { setLoading, unsetLoading } from '../formState/actions';

export const SET_USER = 'SET_USER';
export const SET_SIGN_UP_SERVER_ERRORS = 'SET_SIGN_UP_SERVER_ERRORS';
export const SET_SIGN_IN_SERVER_ERRORS = 'SET_SIGN_IN_SERVER_ERRORS';

export function setUser(user) {
  return dispatch => dispatch({
    type: SET_USER,
    id: user.id,
    name: user.name,
    role: user.role,
    email: user.email,
  });
}

function setSignUpErrors(errors) {
  return dispatch => dispatch({
    type: SET_SIGN_UP_SERVER_ERRORS,
    errors,
  });
}

function setSignInErrors(errors) {
  return dispatch => dispatch({
    type: SET_SIGN_IN_SERVER_ERRORS,
    errors,
  });
}

export function postUser(user) {
  return (dispatch) => {
    dispatch(setLoading());
    axios.post('api/v1/auth/signUp', user).then((response) => {
      localStorage.setItem('id', response.data.id);
      localStorage.setItem('email', response.data.email);
      localStorage.setItem('name', response.data.fullName);
      localStorage.setItem('role', response.data.role);
      localStorage.setItem('token', response.data.token);
      setTimeout(() => {
        window.location = '/';
      }, 500);
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setSignUpErrors(error.response.data));
    });
  };
}


export function signInUser(user) {
  return (dispatch) => {
    dispatch(setLoading());
    axios.post('api/v1/auth/signIn', user).then((response) => {
      localStorage.setItem('id', response.data.id);
      localStorage.setItem('email', response.data.email);
      localStorage.setItem('name', response.data.fullName);
      localStorage.setItem('role', response.data.role);
      localStorage.setItem('token', response.data.token);
      setTimeout(() => {
        window.location = '/';
      }, 500);
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setSignInErrors(error.response.data));
    });
  };
}
