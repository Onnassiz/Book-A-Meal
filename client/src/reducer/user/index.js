import { SET_USER, SET_SIGN_UP_SERVER_ERRORS, SET_SIGN_IN_SERVER_ERRORS } from './action';

const initialState = {};

export default (state = initialState, action) => {
	switch (action.type) {
	case SET_USER:
		return { name: action.name, email: action.email, role: action.role };
	case SET_SIGN_UP_SERVER_ERRORS:
		return { signUpErrors: action.errors };
	case SET_SIGN_IN_SERVER_ERRORS:
		return { signInErrors: action.errors };
	default:
		return state;
	}
};
