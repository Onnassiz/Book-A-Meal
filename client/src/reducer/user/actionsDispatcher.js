import axios from 'axios';
import { setLoading, unsetLoading } from '../formState/actionsDispatcher';
import { setUserAction, setSignUpErrorsAction, setSignInErrorsAction } from './actions';

export const setUser = user => dispatch => dispatch(setUserAction(user));

const setSignUpErrors = errors => dispatch => dispatch(setSignUpErrorsAction(errors));
const setSignInErrors = errors => dispatch => dispatch(setSignInErrorsAction(errors));

const processSignIn = (data) => {
  localStorage.setItem('id', data.id);
  localStorage.setItem('email', data.email);
  localStorage.setItem('name', data.fullName);
  localStorage.setItem('role', data.role);
  localStorage.setItem('token', data.token);
  setTimeout(() => {
    window.location = '/';
  }, 500);
};

export const postUser = user => (dispatch) => {
  dispatch(setLoading());
  return axios.post('api/v1/auth/signUp', user).then((response) => {
    dispatch(unsetLoading());
    processSignIn(response.data);
  }).catch((error) => {
    dispatch(setSignUpErrors(error.response.data));
    dispatch(unsetLoading());
  });
};


export const signInUser = user => (dispatch) => {
  dispatch(setLoading());
  return axios.post('api/v1/auth/signIn', user).then((response) => {
    dispatch(unsetLoading());
    processSignIn(response.data);
  }).catch((error) => {
    dispatch(setSignInErrors(error.response.data));
    dispatch(unsetLoading());
  });
};
