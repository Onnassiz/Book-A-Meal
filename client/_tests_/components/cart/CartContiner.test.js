import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import user from '../../objectProps/user.props';
import CartContainer from '../../../src/components/container/CartContainer';
import getMealsMock from '../../../../testHelpers/meals/mock';

const mockStore = configureStore();

const store = mockStore({
  orders: {},
  formState: { isLoading: false },
  user,
  cart: {
    cart: getMealsMock(1),
    owner: '',
    orderId: '',
    address: '',
    contact: '',
    isOpen: false,
    totalPrice: 0,
    updateMode: false,
  },
});

const wrapper = shallow(<CartContainer store={store} />);

describe('<CartContainer />', () => {
  it('Should render the CartContainer container successfully', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
