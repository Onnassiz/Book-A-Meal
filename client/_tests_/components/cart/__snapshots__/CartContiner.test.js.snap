// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartContainer /> Should render the CartContainer container successfully 1`] = `
<Cart
  cart={
    Object {
      "address": "",
      "cart": Array [
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 0,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 40,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 80,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 120,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 160,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 200,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 240,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 280,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 320,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 360,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 400,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 440,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 480,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 520,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 560,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 600,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 640,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 680,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 720,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 760,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 800,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 840,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 880,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 920,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 960,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 1000,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 1040,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 1080,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 1120,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 1160,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 1200,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 1240,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 1280,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 1320,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 1360,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 1400,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 1440,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 1480,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 1520,
          "userId": 1,
        },
        Object {
          "category": "dolor sit",
          "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
          "name": "Malvaceae",
          "price": 1560,
          "userId": 1,
        },
      ],
      "contact": "",
      "isOpen": false,
      "orderId": "",
      "owner": "",
      "totalPrice": 0,
      "updateMode": false,
    }
  }
  deleteFromCart={[Function]}
  emptyCart={[Function]}
  formState={
    Object {
      "isLoading": false,
    }
  }
  orders={Object {}}
  postOrder={[Function]}
  setCartState={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  updateCart={[Function]}
  updateOrder={[Function]}
  user={
    Object {
      "email": "onnassiz@gmail",
      "id": "3a3c499f-803a-4e06-a328-307daff22243",
      "name": "Ben Onah",
      "role": "caterer",
    }
  }
/>
`;
